src/V3Width.cpp

```diff
@@ -1337,11 +1353,28 @@ class WidthVisitor final : public VNVisitor {
         VL_DO_DANGLING(nodep->deleteTree(), nodep);
     }
     virtual void visit(AstEventControl* nodep) override {
-        nodep->v3warn(E_UNSUPPORTED, "Unsupported: event control statement in this location\n"
-                                         << nodep->warnMore()
-                                         << "... Suggest have one event control statement "
-                                         << "per procedure, at the top of the procedure");
-        VL_DO_DANGLING(nodep->unlinkFrBack()->deleteTree(), nodep);
+        if (nodep->fileline()->timingOn()) {
+            if (v3Global.opt.timing()) {
+                if (VN_IS(m_ftaskp, Func)) {
+                    nodep->v3error("Event controls are not legal in functions. Suggest use a task "
+                                   "(IEEE 1800-2017 13.4.4)");
+                    VL_DO_DANGLING(nodep->unlinkFrBack()->deleteTree(), nodep);
+                    return;
+                }
```

Comment:
I think this error should be given even without `v3Global.opt.timing()` as it's illegal input according to the standard (unless other tools accept this, in which case we should still throw a warning). @wsnyder thoughts?

Reply:
I agree we should always throw this message, please also have the test.pl intentionally use --no-timing to make sure we don't loose that fact.